towhee/hub/file_manager.py

```diff
@@ -311,13 +311,12 @@ def get_pipeline(self, pipeline: str, tag: str = 'main', install_reqs=True):
             author = pipeline_split[0]
             repo = pipeline_split[1]
             file_name = repo.replace('-', '_')
-
             pipeline_path = self._cache_name(repo, author, tag) / (file_name + '.yaml')
-
             file_path = self._config.default_cache / 'pipelines' / pipeline_path
-            found_existing = False
 
-            for path in self._config.cache_paths:
+            found_existing = bool(file_path.is_file())
+
+            for path in reversed(self._config.cache_paths):
```

Comment:
Adding a cache path adds it to the front of the list so reverse can be avoided, this change would look through in the wrong order.

Reply:
I thought we are searching from the default cache path.

Reply:
for path in reversed(self._config.cache_paths):
Was talking about this part, dont need the reversed. Cache path is searched in opposite order added with the default being the last one. If user adds c1, c2, c3, cache paths list would be [c3, c2, c1] with default being c1. Default is the last checked.